### PR TITLE
Adds Nuke Ops gamemode

### PR DESCRIPTION
**Pull request by badbon**
### Purpose
Adds nuke ops gamemode as described in #972 

Here's snippet from round end conditions to give basic idea of this PR:

https://i.gyazo.com/7387653789d868fa211b04c6bcf05c32.png

Round total time is 10 minutes, in 8 minutes (when 2 minutes are left) shuttle spawns at random position in space and heads towards station docking spot, this takes up to 1 minute, sometimes less. When it arrives there is announcement ("Escape shuttle arrived, crew has 1 minute to get on" etc), and at 1 minute left mark, shuttle departs.

If at least 1 crew member manages to escape alive, Syndicate loses.

Let me know if I missed anything or anything else needs to be done.

Thanks Foma for constant support and fixes with matrixmove aspect of this PR.

### Open Questions and Pre-Merge TODOs

- [ ]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [x]  This PR has been tested in multiplayer

### Notes:
Respawns are still functioning as normal, tell me if this is wrong and I'll fix it before merge. 

Also, if its possible merge my PR this time unlike last time when Doobly had to recreate PR on his own because of some merge conflict, I like when activity appears on my profile :D 

As for last note, scene and managers prefab was both changed so look out for that.

Thanks.

EDIT: Oh, and I tried making this function on flashlight and I added nuke there but getting shuttle there was a bit messy so I'll leave it to someone else, it should be very simple to implement my code into any map if the map has proper docking spot.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Shuttles/MatrixMove.cs [hunk 1]
```diff
@@ -475,7 +482,12 @@ public void RotateTo( Orientation desiredOrientation )
 	}
 
 	///Zero means 100% accurate, but will lead to peculiar behaviour (autopilot not reacting fast enough on high speed -> going back/in circles etc)
-	private static readonly int AccuracyThreshold = 1;
+	private static int AccuracyThreshold = 1;
+
+	public void SetAccuracy(int newAccuracy)
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Shuttles/MatrixMove.cs:487]: Smart move :)

File: UnityProject/Assets/Scripts/Shuttles/EscapeShuttle.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	IEnumerator ReverseIntoStation(MatrixMove mm)
+	{
+		yield return new WaitForSeconds(3f);
+		mm.MoveFor(offSetReverse);
+	}
+
+	public void SpawnNearStation()
+	{
+		//Picks a slightly random position for shuttle to spawn in to try avoid interception from syndicate
+		GetComponent<MatrixMove>().SetPosition(Random.insideUnitCircle * 500 + new Vector2(500, -500));
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Shuttles/EscapeShuttle.cs]: Why not use `mm` here?
You probably want to round the result of that `Random.insideUnitCircle * 500` by wrapping it in `Vector2Int.RoundToInt()`, I don't think `transform.position.Normalize()` would do any good as transform represents clientside lerping and is likely off

Also I'm curious if shuttle's arrival time is constant or varies depending on that bluespace jump position?

File: UnityProject/Assets/Scripts/Shuttles/EscapeShuttle.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+		transform.position.Normalize();
+
+		spawnedIn = true;
+		ApproachStation();
+	}
+
+	public void ApproachStation()
+	{
+		GetComponent<MatrixMove>().SetSpeed(25);
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Shuttles/EscapeShuttle.cs]: here as well

File: UnityProject/Assets/Scripts/Managers/PlayerList.cs [hunk 4]
```diff
@@ -99,30 +99,71 @@ public void TryAddScores(string uniqueName)
...
+	}
 
-		var scoreSort = departmentScores.OrderByDescending(pair => pair.Value)
-			.ToDictionary(pair => pair.Key, pair => pair.Value);
+	public int GetCrewAliveCount()
+	{
+		int alive = 0;
 
-		foreach ( KeyValuePair<JobDepartment, int> ds in scoreSort )
+		PlayerScript[] alivePlayers = FindObjectsOfType<PlayerScript>();
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Managers/PlayerList.cs]: You're better off using InGamePlayers, it's more convenient and efficient. See example that prints player positions:
```cs
		foreach (ConnectedPlayer player in PlayerList.Instance.InGamePlayers) {
			//Printing this the pretty way, example:
			//Bob (CAPTAIN) is located at (77,0, 52,0, 0,0)
			Logger.Log( $"{player.Name} ({player.Job}) is located at {player.Script.WorldPos}" );
		}
```
you can avoid GetComponent and FindObjectsOfType this way

File: UnityProject/Assets/Scripts/Shuttles/EscapeShuttle.cs [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+	}
+
+	public int GetCrewCountOnboard() //Returns how many crew members (excluding syndicate) are on the shuttle
+	{									// (Used to calculate if crew managed to escape at end of round)
+		int crewCount = 0;
+		PlayerScript[] crewMembers = FindObjectsOfType<PlayerScript>();
+
+		foreach(PlayerScript ps in crewMembers)
+		{
+			if (ps.JobType != JobType.SYNDICATE && ps.gameObject.GetComponent<PlayerHealth>().Health > 0 && ps.gameObject.transform.root == gameObject.transform.root)
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Shuttles/EscapeShuttle.cs]: Use `PlayerList.Instance.InGamePlayers` here to get crew members.

Also that transform.root trick is interesting, but relying on transforms isn't good in the long run: they reflect slightly delayed clientside data and you should use the freshest server data. Compare matrix IDs instead:
`player.Script.PlayerSync.ServerState.MatrixID == mm.MatrixInfo.Id`

**Review by fomalsd — COMMENTED** [review]
Looks good!

File: UnityProject/Assets/Scripts/Shuttles/EscapeShuttle.cs (hunk 2 above)
Comment (badbon) [reply UnityProject/Assets/Scripts/Shuttles/EscapeShuttle.cs]: Yeah I missed mms, fixed.

Also after some testing .Normalize or any other solution isn't necessary so I'll completely remove those.

File: UnityProject/Assets/Scripts/Managers/PlayerList.cs (hunk 4 above)
Comment (badbon) [reply UnityProject/Assets/Scripts/Managers/PlayerList.cs]: Thanks, fixed.

**Pull request merged**